comet-java-client/src/main/java/ml/comet/experiment/impl/BaseExperimentAsync.java

```diff
@@ -334,13 +344,55 @@ void uploadAsset(@NonNull File file, @NonNull String fileName,
         this.logAsset(asset, onComplete);
     }
 
+    /**
+     * Asynchronous version that only logs any received exceptions or failures.
+     *
+     * @param uri        the {@link URI} pointing to the remote asset location. There is no imposed format,
+     *                   and it could be a private link.
+     * @param fileName   the optional "name" of the remote asset, could be a dataset name, a model file name.
```

Comment:
Yes, the `fileName` can be skipped by user and it would be taken from the `URI`. This is the same behaviour as in Python SDK.

However, the method you are referring is for internal purposes only and not exposed as a public API. The public API methods has overloads without `fileName` parameter.